friends/src/App.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import './App.css';
+import axios from 'axios';
+import Friend from './components/friends'
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      friends: [],
+      name: '',
+      age: '',
+      email: '',
+      id: ''
+    }
+  }
+
+  componentDidMount() {
+    axios
+      .get('http://localhost:5000/friends')
+      .then(response => {
+        console.log(response)
+        this.setState({ friends: response.data })
+      })
+  }
+  onChangeHandler = e => {
+    const name = e.target.name;
+    this.setState({ [name]: e.target.value })
+  }
+
+  addFriendHandler = (e) => {
+    e.preventDefault();
+    axios.post("http://localhost:5000/friends", {
+      name: this.state.name,
+      age: this.state.age,
+      email: this.state.email
+
+    })
+      .then(response => {
+        console.log(response)
+        this.setState({ friends: response.data })
+      })
+      .catch(err => console.log(err))
+  }
+
+  deleteFriend = (id) => {
+    return () => {
+      axios.delete(`http://localhost:5000/friends/${id}`)
+        .then(response => {
+          this.setState({ friends: response.data })
+        })
+        .catch(err => console.log(err))
+    }
+  }
+
+  updateFriend = (e, id, name, age, location) => {
+    id = this.state.id
+    e.preventDefault()
+    axios.put(`http://localhost:5000/friends/${id}`,
+      {
+        name: this.state.name,
+        age: this.state.age,
+        email: this.state.email
+      }
+    )
+      .then(response => {
+        this.setState({ friends: response.data })
+      })
+  }
+
+  render() {
```

Comment:
Destructuring state here would let you replace every usage of `this.state.prop` with `prop`. 
`const { name, age, email } = this.state` , just a technique to file away when our data starts to get a little bit more complex.